core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -4292,6 +4292,37 @@ private Sql checkSubQuery(String sql) {
     checkPlanning(program, sql);
   }
 
+  @Test public void testReduceAverageWithNoReduceSum() {
+    final AggregateReduceFunctionsRule.AggregateFunctionsToReduce functionsToReduce =
+        new AggregateReduceFunctionsRule.AggregateFunctionsToReduce(
+        true, false, true,
+        true, true, true,
+        true, true, true,
+        true);
+    checkPlanning(
+        AggregateReduceFunctionsRule.getInstanceWithSpecificFunctionsToReduce(
+          functionsToReduce),
+                  "select name, max(name), avg(deptno), min(name)"
+                          + " from sales.dept group by name");
+  }
+
+  @Test public void testNoReduceSum() {
```

Comment:
If it is not too much effort can you add a test for each relevant function? Possibly a `@RunWith(Parameterized.class)` test?

Reply:
I have added more tests. Please let me know if the new tests are not added the way you were thinking/suggesting.

Reply:
done.